libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -1647,7 +1667,11 @@ bool AP_AHRS_NavEKF::attitudes_consistent(char *failure_msg, const uint8_t failu
     // for EKF3. DCM can't use external yaw, so we don't expect it's
     // yaw to align with EKF3 when EKF3 is using an external yaw
     // source
-    if (ekf_type() != EKFType::THREE || !EKF3.using_external_yaw()) {
+    bool using_external_yaw = false;
+#if HAL_NAVEKF3_AVAILABLE
+    using_external_yaw = ekf_type() == EKFType::THREE && EKF3.using_external_yaw();
```

Comment:
Should this be `active_ekf_type()`?  It's pre-existing.

Reply:
not, it should be ekf_type(). ek3 may not be active yet